test/functional/p2p_tx_download.py

```diff
@@ -216,6 +216,24 @@ def test_preferred_inv(self):
         with p2p_lock:
             assert_equal(peer.tx_getdata_count, 1)
 
+    def test_txid_inv_delay(self):
+        self.log.info('Check that inv from a txid-relay peers are delayed by {} s'.format(TXID_RELAY_DELAY))
+        self.restart_node(0, extra_args=['-whitelist=noban@127.0.0.1'])
+        mock_time = int(time.time() + 1)
+        self.nodes[0].setmocktime(mock_time)
+        peer = self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=False))
+        # Add a second wtxid-relay connection otherwise TXID_RELAY_DELAY is waived in
+        # lack of wtxid-relay peers
+        self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=True))
+        peer.send_message(msg_inv([CInv(t=MSG_TX, h=0xff11ff11)]))
+        peer.sync_with_ping()
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 0)
+        self.nodes[0].setmocktime(mock_time + TXID_RELAY_DELAY)
```

Comment:
Shouldn't `NONPREF_PEER_TX_DELAY` also be applied here?

Anyway, I'd first jump in `NONPREF_PEER_TX_DELAY` to see that it's not enough time passed, and then jump `TXID_RELAY_DELAY` again to see that enough time passed? I think that's what is supposed to happen..

Reply:
The positive case was already tested in `test_preferred_inv` but added a mutation to cover the negative one, i.e applying `NONPREF_PEER_TX_DELAY` on non-preferred peers.